### PR TITLE
695: Add --format Flag to "zally rules" Command

### LINKED ISSUE #695: Add --format flag to zally rules command
In scope of #652 we have added markdown format support for violations output. Let's add the same `--format` flag to the `zally rules` command:

```bash
zally rules --format markdown
zally rules --format pretty
```

Note: formatters already support `FormatRule`. Steps to be done:
- Make `--format` flag global
- Use a proper formatter in `zally rules`

### PR DESCRIPTION
**Pull request by netme**
Closes #695

### CODE REVIEW

File: cli/zally/utils/formatters/factories.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package formatters
+
+import (
+	"fmt"
+)
+
+// NewFormatter creates a formatter for a given format
+func NewFormatter(format string) (Formatter, error) {
+
+	if format == "markdown" {
```
Comment (maxim-tschumak) [inline cli/zally/utils/formatters/factories.go]: A switch statement would be nicer here.

**Comment by maxim-tschumak** [comment]
:+1:

File: cli/zally/utils/formatters/factories.go (hunk 1 above)
Comment (netme) [reply cli/zally/utils/formatters/factories.go]: Addressed in 711a751

**Comment by netme** [comment]
@maxim-tschumak could you please check it again?

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by netme** [comment]
:+1:

**Pull request merged**